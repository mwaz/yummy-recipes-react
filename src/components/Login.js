import React, { Component } from 'react';
import { Redirect } from 'react-router-dom'
import axios from 'axios'
import Navbar from './navbar'
import Footer from './footer'
import { Panel, Col, FormGroup, FormControl, Button, InputGroup } from 'react-bootstrap';

class Login extends Component {
    constructor(props) {
        super(props)
        this.state = {
            email: '',
            password: '',
            redirect : '',

        }
    }
    handleClick(event){
        const url = 'http://127.0.0.1:5000/yummy_api/v1/auth';
        let payload = {
            'email': this.state.email,
            'password': this.state.password,
        }
        axios.post(url + '/login ', payload)
            .then((response) => {
                console.log(response.data.message),
                    this.setState({ redirect: true })
            });

        }
    
    render() {
        const redirect = this.state.redirect
        if (redirect) {
            return <Redirect to={{ pathname: '/categories' }} />
        }
        return (
            <div>
                <Navbar />
            <div className="background-div">
                <div className="background-container">
                    <div className="background-container-form">
                        <center><strong> <div style={{ fontSize: "20px", paddingBottom: "1%" }}>Login  Here </div></strong></center>
                        <Panel header='Register' bsStyle="warning">
                            <FormGroup>
                                <FormControl style={{ backgroundColor: "black", color: "white", filter: "opacity(1)", }} type="email" id="email" placeholder="email" onChange={(event) => this.setState({ email: event.target.value })} />
                            </FormGroup>
                            <FormGroup>
                                <FormControl style={{ backgroundColor: "black", color: "white", filter: "opacity(1)", }} type="password" id="password" placeholder="Password" onChange={(event) => this.setState({ password: event.target.value })} />
                            </FormGroup>
                            <Button bsStyle="success" onClick={(event => this.handleClick(event))}>Login</Button>
                        </Panel>
                    </div>
                </div>
            </div>
            <Footer />
            </div>

        );
    }
}
export default Login;
