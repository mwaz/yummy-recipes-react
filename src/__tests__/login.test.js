import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Login from '../components/auth-front/login';
import toJson, { shallowToJson } from 'enzyme-to-json';

describe('Categories component', () => {
    const login = jest.fn();
    const wrapper = shallow(<Login login={login} location={{}} />);
    const preventDefault = jest.fn();

    it('renders properly without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('submits data', () => {
        wrapper.find('Button').simulate('submit');
        
    });

    it('it renders state initially', () => {
        expect(wrapper.state().email).toEqual('');
        expect(wrapper.state().password).toEqual('');
    });

    it('should login a user', () => {
        const component = shallow(<Login />);
        const preventDefault = jest.fn();
        component.setState({
            email: 'hey@hey.com',
            password: '1234567',
        });
        component.find('Button').simulate('submit', { preventDefault });
        expect(toJson(component)).toMatchSnapshot();
    });
    
    
});