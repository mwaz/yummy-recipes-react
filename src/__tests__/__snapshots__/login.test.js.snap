// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component renders properly without crashing 1`] = `
<div>
  <div
    className="background-div"
  >
    <NavigationBar />
    <div
      className="background-container"
    >
      <div
        className="background-container-form"
      >
        <center>
          <strong>
             
            <div
              style={
                Object {
                  "fontSize": "20px",
                  "paddingBottom": "1%",
                }
              }
            >
              Login  Here 
            </div>
          </strong>
        </center>
        <ToastContainer
          autoClose={5000}
          bodyClassName=""
          className={null}
          closeButton={
            <DefaultCloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnHover={true}
          position="top-right"
          progressClassName=""
          style={null}
          toastClassName=""
          transition={[Function]}
        />
        <div
          className="card"
        >
          <div
            className="card-title"
          >
            Login 
          </div>
          <div
            className="card-text"
          >
            <Uncontrolled(Panel)
              bsStyle="warning"
              header="Register"
            >
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="email"
                  onChange={[Function]}
                  placeholder="email"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="email"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="password"
                  onChange={[Function]}
                  placeholder="Password"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="password"
                />
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="success"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Login
              </Button>
            </Uncontrolled(Panel)>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Categories component should login a user 1`] = `
<div>
  <div
    className="background-div"
  >
    <NavigationBar />
    <div
      className="background-container"
    >
      <div
        className="background-container-form"
      >
        <center>
          <strong>
             
            <div
              style={
                Object {
                  "fontSize": "20px",
                  "paddingBottom": "1%",
                }
              }
            >
              Login  Here 
            </div>
          </strong>
        </center>
        <ToastContainer
          autoClose={5000}
          bodyClassName=""
          className={null}
          closeButton={
            <DefaultCloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnHover={true}
          position="top-right"
          progressClassName=""
          style={null}
          toastClassName=""
          transition={[Function]}
        />
        <div
          className="card"
        >
          <div
            className="card-title"
          >
            Login 
          </div>
          <div
            className="card-text"
          >
            <Uncontrolled(Panel)
              bsStyle="warning"
              header="Register"
            >
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="email"
                  onChange={[Function]}
                  placeholder="email"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="email"
                />
              </FormGroup>
              <FormGroup
                bsClass="form-group"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  id="password"
                  onChange={[Function]}
                  placeholder="Password"
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "black",
                      "filter": "opacity(1)",
                    }
                  }
                  type="password"
                />
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="success"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Login
              </Button>
            </Uncontrolled(Panel)>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
